pkg/controller/operator/common/defaults.go

```diff
@@ -322,6 +322,14 @@ func DefaultConfiguration(config *operatorv1alpha1.KubermaticConfiguration, logg
 
 	copy := config.DeepCopy()
 
+	if copy.Spec.ImagePullSecret == "" {
+		// This ensures that the created dockercfg Secrets later will contain valid JSON
+		// and that we do not have to make a distinction between unset, empty and full
+		// Docker Pull Secret; instead we *always* create a dockercfg Secret and reference
+		// it everywhere.
+		copy.Spec.ImagePullSecret = "{}"
+	}
+
```

Comment:
Do we need this when we have [this](https://github.com/kubermatic/kubermatic/pull/5844/files#diff-bdbc7758829c9b98e5794ed1f89bffd5R23)?

Reply:
Yes, because those are different secrets. The one in the Kubermatic Operator chart is for pulling the _operator_ only, and then you have another ImagePullSecret in the KubermaticConfiguration, which is deployed by the operator to pull Kubermatic itself.

It's ugly, but an unfortunate side effect of our Helm chart structure at the moment.

Reply:
Please revert this as we agreed, empty `imagePullSecret` will be covered in #5874